api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -112,6 +112,23 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 		}
 	}
 
+	private Obs check_patient(Obs obs) {
```

Comment:
I didn't give the function an appropriate name. ill  work on it. What the function does is that it takes an obs as input and it checks whether the obs.person id and its encounters patient id match.